### PR TITLE
test: use `T.TempDir` to create temporary test directory

### PR DESCRIPTION
**Pull request by Juneezee**
### What does this PR do?

A testing cleanup. 

This pull request replaces `ioutil.TempDir` with `t.TempDir`. We can use the `T.TempDir` function from the `testing` package to create temporary directory. The directory created by `T.TempDir` is automatically removed when the test and all its subtests complete. 

Reference: https://pkg.go.dev/testing#T.TempDir

```go
func TestFoo(t *testing.T) {
	// before
	tmpDir, err := ioutil.TempDir("", "")
	require.NoError(t, err)
	defer os.RemoveAll(tmpDir)

	// now
	tmpDir := t.TempDir()
}
```

### Motivation

This saves us at least 2 lines (error check, and cleanup) on every instance, or in some cases adds cleanup that we forgot.

### Describe how to test/QA your changes

All tests should continue to pass. No leftover temporary directories after the tests finish.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ameske — APPROVED** [review]
👍🏻 for Remote Config

File: pkg/util/containers/providers/cgroup/proc_test.go
Comment (djmitche) [inline pkg/util/containers/providers/cgroup/proc_test.go:19]: It looks like the first return value is never used?

**Review by djmitche — CHANGES_REQUESTED** [review]
This seems to mix a few large-scale changes:
 * using T.TempDir()
 * closing files returned from os.Create (and indirectly by ioutil.TempFile)

In general, when making wide-ranging but formulaic changes like this, it's easier to review if each type of change is in its own commit, as it makes the repetition clear.  In this case, I could scan quickly for `.. = ioutil.TempDir(..)` -> `t.TempDir()`, then at the next commit look at the `os.Create` calls.  

That said, this PR looks great -- just a few tiny bits to update.  Thanks for this!

**Review by paulcacheux — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

File: pkg/flare/archive_test.go
Comment (djmitche) [inline pkg/flare/archive_test.go:289]: Looks like this got duplicated in the fix-up :(

**Review by djmitche — APPROVED** [review]
Looks fine, just one redundant line to remove..

**Review by ahmed-mez — APPROVED** [review]

**Comment by trishankatdatadog** [comment]
@cedricvanrompay-datadog seems relevant to SIT

**Comment by djmitche** [comment]
> Can someone please remove me specifically from reviewing this PR? Thanks!

You're in CODEOWNERS here:
```
/cmd/agent/app/integrations*.go         @trishankatdatadog @DataDog/agent-integrations @DataDog/agent-core
```
should we remove you from that line in CODEOWNERS?  Otherwise, to remove yourself from just this PR you can submit an empty comment.

**Comment by paulcacheux** [comment]
Let's merge https://github.com/DataDog/datadog-agent/pull/12221 then

**Review by ajgajg1134 — APPROVED** [review]
👍 for agent-apm, looks good!

**Comment by trishankatdatadog** [comment]
> Let's merge #12221 then

Yes please 🙏🏽

**Review by robertjli — APPROVED** [review]

**Comment by Juneezee** [comment]
Rebased onto `main` and resolved conflicts in `pkg/persistentcache/persistentcache_test.go`.

**Review by ofek — APPROVED** [review]
💯

**Comment by sgnn7** [comment]
@Juneezee Seems like there's a conflict in one of the files as of today. Since there's also a cut-off from 7.38 branch today, we may need to change the milestone to 7.39 instead.

@trishankatdatadog / @cedricvanrompay-datadog Can one of you (or someone else from @DataDog/software-integrity-and-trust) review this PR since that approval is still missing from it?

**Review by trishankatdatadog — APPROVED** [review]

**Comment by Juneezee** [comment]
Rebased onto `main` and resolved conflicts.

**Comment by olivielpeau** [comment]
Changed the milestone to 7.39 since this didn't make it in time for 7.38. We can merge this once the 7.38 code freeze ends.

**Comment by sgnn7** [comment]
@Juneezee Merged! Thank you for your contribution (and apologies for it taking a bit longer than expected)!

**Pull request merged**
